server/conn.go

```diff
@@ -2090,7 +2090,20 @@ func (cc *clientConn) handleStmt(ctx context.Context, stmt ast.StmtNode, warns [
 	ctx = context.WithValue(ctx, util.ExecDetailsKey, &util.ExecDetails{})
 	reg := trace.StartRegion(ctx, "ExecuteStmt")
 	cc.audit(plugin.Starting)
-	rs, err := cc.ctx.ExecuteStmt(ctx, stmt)
+	var rs ResultSet
+	var err error
+	rs, err = cc.ctx.ExecuteStmt(ctx, stmt)
+	cs, ok := cc.ctx.GetSessionVars().GetCache(stmt)
+	if !ok {
+		if variable.ResultCacheSize.Load() != 0 {
```

Comment:
Check the variable first, and then `cc.ctx.GetSessionVars().GetCache(stmt)`

This can avoid the cost of normal code logic when the variable is disabled.

Reply:
GetCache Include variable detection.
<img width="605" alt="image" src="https://user-images.githubusercontent.com/97376271/212100604-a64223e8-e3e4-4ed3-b114-9f988d10e274.png">